pelican/tests/test_contents.py

```diff
@@ -497,6 +497,30 @@ def test_slugify_tags_with_dots(self):
         article = Article(**article_kwargs)
         self.assertEqual(article.url, 'fedora.qa/this-week-in-fedora-qa/')
 
+    def test_check_save_as_detects_breakout(self):
+        settings = get_settings()
+        settings['TAG_SUBSTITUTIONS'] = [('Fedora QA', 'fedora.qa', True)]
+        settings['ARTICLE_URL'] = '{tag}/{slug}/'
+        article_kwargs = self._copy_page_kwargs()
+        article_kwargs['metadata']['tag'] = Tag('Fedora QA', settings)
```

Comment:
Actually, I have another issue with this test. You're defining `ARTICLE_URL` but testing for `ARTICLE_SAVE_AS` (which is left as default `{slug}`), and technically they are not directly related. This is like trying to intentionally confuse developers. First time I looked at this, it was "Wait, what? Why is this `False`?" :).